internal/cmd/skupper/site/kube/site_update.go

```diff
@@ -0,0 +1,242 @@
+	Client             skupperv1alpha1.SkupperV1alpha1Interface
+	CobraCmd           cobra.Command
+	flags              UpdateFlags
+	options            map[string]string
+	siteName           string
+	serviceAccountName string
+	Namespace          string
+	linkAccessType     string
+	output             string
+}
+
+func NewCmdSiteUpdate() *CmdSiteUpdate {
+
+	skupperCmd := CmdSiteUpdate{}
+
+	cmd := cobra.Command{
+		Use:     "update <name>",
+		Short:   "Change site settings",
+		Long:    siteUpdateLong,
+		Example: "skupper site update my-site --enable-link-access",
+		PreRun:  skupperCmd.NewClient,
+		Run: func(cmd *cobra.Command, args []string) {
+			utils.HandleErrorList(skupperCmd.ValidateInput(args))
+			skupperCmd.InputToOptions()
+			utils.HandleError(skupperCmd.Run())
+		},
+		PostRunE: func(cmd *cobra.Command, args []string) error {
+			return skupperCmd.WaitUntilReady()
+		},
+	}
+
+	skupperCmd.CobraCmd = cmd
+	skupperCmd.AddFlags()
+
+	return &skupperCmd
+}
+
+func (cmd *CmdSiteUpdate) NewClient(cobraCommand *cobra.Command, args []string) {
+	cli, err := client.NewClient(cobraCommand.Flag("namespace").Value.String(), cobraCommand.Flag("context").Value.String(), "")
+	utils.HandleError(err)
+
+	cmd.Client = cli.GetSkupperClient().SkupperV1alpha1()
+	cmd.Namespace = cli.Namespace
+}
+
+func (cmd *CmdSiteUpdate) AddFlags() {
+	cmd.CobraCmd.Flags().BoolVar(&cmd.flags.enableLinkAccess, "enable-link-access", false, "allow access for incoming links from remote sites (default: false)")
+	cmd.CobraCmd.Flags().StringVar(&cmd.flags.linkAccessType, "link-access-type", "", `configure external access for links from remote sites.
+Choices: [route|loadbalancer]. Default: On OpenShift, route is the default; 
+for other Kubernetes flavors, loadbalancer is the default.`)
+	cmd.CobraCmd.Flags().StringVar(&cmd.flags.serviceAccount, "service-account", "skupper-controller", "the Kubernetes service account under which to run the Skupper controller")
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.output, "output", "o", "", "print resources to the console instead of submitting them to the Skupper controller. Choices: json, yaml")
+}
+
+func (cmd *CmdSiteUpdate) ValidateInput(args []string) []error {
+
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	linkAccessTypeValidator := validator.NewOptionValidator(utils.LinkAccessTypes)
+	outputTypeValidator := validator.NewOptionValidator(utils.OutputTypes)
+
+	//Validate if there is already a site defined in the namespace
+	siteList, _ := cmd.Client.Sites(cmd.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if siteList != nil && len(siteList.Items) == 0 {
+		validationErrors = append(validationErrors, fmt.Errorf("there is no existing Skupper site resource to update"))
+	} else if len(siteList.Items) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("there are several sites in this namespace and and it should be only one"))
```

Comment:
same comment as for the site delete... if you specify the site name, then this should not be thrown